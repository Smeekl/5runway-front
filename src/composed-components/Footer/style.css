.footer {
  font-weight: 300;
  font-size: 16px;
  padding: 0 13.5% 2% 11.5%;
  padding-top: 50px;
  background-color: var(--color-primary);
}

.social {
  width: 52%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding-bottom: 2%;
}

.footer__menu {
  height: 310px;
  display: flex;
  flex-direction: column;
  align-content: space-between;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.menu {
  display: flex;
  flex-direction: column;
}

.list__link {
  color: var(--color-secondary-white);
  transition: 0.2s;
}

.list__item {
  margin-bottom: 13px;
}

.list__link:hover {
  color: var(--color-secondary-white);
  opacity: 0.7;
}

.copyright {
  font-size: 12px;
  display: flex;
  flex-direction: column;
  color: var(--color-secondary-white);
}

.copyright span {
  color: var(--color-secondary-white);
}

.info {
  color: var(--color-secondary-white);
  justify-content: space-between;
  display: flex;
  flex-direction: row;
}

.footer__icon {
  display: inline-flex;
  transition: 0.2s all;
}

.footer__icon:hover {
  opacity: 0.7;
}

.social__item:hover {
  opacity: 0.7;
}
.social__item {
  transition: 0.2s;
}

.social__link {
  display: inline-flex;
}
