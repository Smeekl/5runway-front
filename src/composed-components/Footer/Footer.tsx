import { Link } from 'react-router-dom';
import { ReactComponent as IconInstagram } from '@svgIcons/iconInstagram.svg';
import { ReactComponent as IconEmail } from '@svgIcons/iconEmail.svg';
import { ReactComponent as IconPinterest } from '@svgIcons/iconPinterest.svg';
import { ReactComponent as IconTwitter } from '@svgIcons/iconTwitter.svg';
import { ReactComponent as IconLinkedIn } from '@svgIcons/iconLinkedIn.svg';
import { ReactComponent as IconFacebook } from '@svgIcons/iconFacebook.svg';
import { ReactComponent as IconSecondLogo } from '@svgIcons/iconSecondLogo.svg';
import './style.css';

const socialMedia = [
  { icon: <IconInstagram />, url: '#' },
  { icon: <IconEmail />, url: '#' },
  { icon: <IconPinterest />, url: '#' },
  { icon: <IconTwitter />, url: '#' },
  { icon: <IconLinkedIn />, url: '#' },
  { icon: <IconFacebook />, url: '#' },
];

const listItems = [
  { name: 'Menu', url: '/menu' },
  { name: 'Profile', url: '/profile' },
  { name: 'Save', url: '/save' },
  { name: 'Compare', url: '/compare' },
  { name: 'Sell now', url: '/sellnow' },
  { name: 'Search', url: '/search' },
  { name: 'Articles', url: '/articles' },
  { name: 'Fevorite products', url: '/fevoriteproducts' },
  { name: 'Aircraft', url: '/aircraft' },
  { name: 'Parts/products', url: '/Parts/products' },
  { name: 'Real estate', url: '/realestate' },
  { name: 'Services', url: '/services' },
  { name: 'Business aircraft', url: '/businessaircraft' },
  { name: 'Campany dealer', url: '/campanydealer' },
  { name: 'Features', url: '/features' },
  { name: 'Engines', url: '/engines' },
  { name: 'Help', url: '/help' },
  { name: 'Security', url: '/security' },
  { name: 'Privacy', url: '/privacy' },
];

const socialMediaList = socialMedia.map(({ icon, url }, index) => (
  <li className="social__item" key={index}>
    <Link to={url} className="social__link">
      {icon}
    </Link>
  </li>
));
const navigationList = listItems.map(({ name, url }, index) => (
  <li className="list__item" key={index}>
    <Link to={url} className="list__link">
      {name}
    </Link>
  </li>
));

const Footer = () => (
  <div className="footer">
    <div className="footer__social social">{socialMediaList}</div>
    <div className="footer__menu menu">{navigationList}</div>
    <div className="footer__info info">
      <div className="info__copyright copyright">
        <span>Runway5</span>
        <span>Copyright Â© 2020</span>
      </div>
      <Link to="/home" className="footer__icon icon  ">
        <IconSecondLogo />
      </Link>
    </div>
  </div>
);

export default Footer;
