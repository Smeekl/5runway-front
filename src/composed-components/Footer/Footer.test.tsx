import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import Footer from './Footer';

describe('Footer  UI tests', () => {
  test('renders Footer', () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('should be in the document', () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
    expect(container).toBeInTheDocument();
  });

  test('navigation menu routes are correct', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Routes>
          <Route path="/menu" element={<div data-testid="menu">Menu</div>} />
          <Route path="/help" element={<div data-testid="help">Help</div>} />
        </Routes>
        <Footer />
      </MemoryRouter>
    );

    const menuLink = getByText('Menu');
    userEvent.click(menuLink);
    expect(screen.getByTestId('menu')).toBeInTheDocument();

    const helpLink = getByText('Help');
    userEvent.click(helpLink);
    expect(screen.getByTestId('help')).toBeInTheDocument();
  });
});
