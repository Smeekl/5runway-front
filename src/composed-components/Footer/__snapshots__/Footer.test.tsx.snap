// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. renders CategoriesMenuCard 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="footer__social social"
    >
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
    </div>
    <div
      class="footer__menu menu"
    >
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/menu"
        >
          Menu
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/save"
        >
          Save
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/compare"
        >
          Compare
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/sellnow"
        >
          Sell now
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/search"
        >
          Search
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/articles"
        >
          Articles
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/fevoriteproducts"
        >
          Fevorite products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/aircraft"
        >
          Aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/Parts/products"
        >
          Parts/products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/realestate"
        >
          Real estate
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/services"
        >
          Services
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/businessaircraft"
        >
          Business aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/campanydealer"
        >
          Campany dealer
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/features"
        >
          Features
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/engines"
        >
          Engines
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/help"
        >
          Help
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/security"
        >
          Security
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/privacy"
        >
          Privacy
        </a>
      </li>
    </div>
    <div
      class="footer__info info"
    >
      <div
        class="info__copyright copyright"
      >
        <span>
          Runway5
        </span>
        <span>
          Copyright © 2020
        </span>
      </div>
      <a
        class="footer__icon icon  "
        href="/home"
      >
        <reactcomponent />
      </a>
    </div>
  </div>
</div>
`;

exports[`ErrorIndicator UI tests 1. renders CategoriesMenuCard 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="footer__social social"
    >
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
    </div>
    <div
      class="footer__menu menu"
    >
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/menu"
        >
          Menu
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/save"
        >
          Save
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/compare"
        >
          Compare
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/sellnow"
        >
          Sell now
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/search"
        >
          Search
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/articles"
        >
          Articles
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/fevoriteproducts"
        >
          Fevorite products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/aircraft"
        >
          Aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/Parts/products"
        >
          Parts/products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/realestate"
        >
          Real estate
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/services"
        >
          Services
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/businessaircraft"
        >
          Business aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/campanydealer"
        >
          Campany dealer
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/features"
        >
          Features
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/engines"
        >
          Engines
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/help"
        >
          Help
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/security"
        >
          Security
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/privacy"
        >
          Privacy
        </a>
      </li>
    </div>
    <div
      class="footer__info info"
    >
      <div
        class="info__copyright copyright"
      >
        <span>
          Runway5
        </span>
        <span>
          Copyright © 2020
        </span>
      </div>
      <a
        class="footer__icon icon  "
        href="/home"
      >
        <reactcomponent />
      </a>
    </div>
  </div>
</div>
`;

exports[`Footer  UI tests renders Footer 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="footer__social social"
    >
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
      <li
        class="social__item"
      >
        <a
          class="social__link"
          href="/"
        >
          <reactcomponent />
        </a>
      </li>
    </div>
    <div
      class="footer__menu menu"
    >
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/menu"
        >
          Menu
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/save"
        >
          Save
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/compare"
        >
          Compare
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/sellnow"
        >
          Sell now
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/search"
        >
          Search
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/articles"
        >
          Articles
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/fevoriteproducts"
        >
          Fevorite products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/aircraft"
        >
          Aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/Parts/products"
        >
          Parts/products
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/realestate"
        >
          Real estate
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/services"
        >
          Services
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/businessaircraft"
        >
          Business aircraft
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/campanydealer"
        >
          Campany dealer
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/features"
        >
          Features
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/engines"
        >
          Engines
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/help"
        >
          Help
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/security"
        >
          Security
        </a>
      </li>
      <li
        class="list__item"
      >
        <a
          class="list__link"
          href="/privacy"
        >
          Privacy
        </a>
      </li>
    </div>
    <div
      class="footer__info info"
    >
      <div
        class="info__copyright copyright"
      >
        <span>
          Runway5
        </span>
        <span>
          Copyright © 2020
        </span>
      </div>
      <a
        class="footer__icon icon  "
        href="/home"
      >
        <reactcomponent />
      </a>
    </div>
  </div>
</div>
`;
